docs/usage.rst

```diff
@@ -0,0 +1,257 @@
+it is recommended to use :class:`audinterface.Feature`,
+which returns a :class:`pandas.DataFrame`
+and assigns names to the dimensions/features.
+
+.. jupyter-execute::
+
+    def features(signal, sampling_rate):
+        return [signal.mean(), signal.std()]
+
+    interface = audinterface.Feature(
+        ['mean', 'std'],
+        process_func=features,
+    )
+
+    df = interface.process_index(index)
+    df
+
+
+Framewise feature interface
+---------------------------
+
+If a processing function does not return
+one set of features for the whole signal,
+but does return features
+in a framewise manner,
+you should specify the ``win_dur``
+and ``hop_dur`` arguments
+of :class:`audinterface.Feature`.
+It's also important the processing function
+returns the value in the correct shape,
+namely ``(num_channels, num_features, num_frames)``,
+whereas the first dimension is optionally.
+
+.. jupyter-execute::
+
+    import librosa
+
+    def features(signal, sampling_rate, win_dur, hop_dur, n_mfcc):
+        hop_length = int(hop_dur * sampling_rate)
+        win_length = int(win_dur * sampling_rate)
+        mfcc = librosa.feature.mfcc(
+            y=signal,
+            sr=sampling_rate,
+            n_mfcc=13,
+            hop_length=hop_length,
+            win_length=win_length,
+        )
+        return mfcc
+
+    win_dur = 0.02
+    hop_dur = 0.01
+    n_mfcc = 13
+    interface = audinterface.Feature(
+        [f'mfcc-{idx}' for idx in range(n_mfcc)],
+        process_func=features,
+        process_func_args={
+            'win_dur': win_dur,
+            'hop_dur': hop_dur,
+            'n_mfcc': n_mfcc,
+        },
+        hop_dur=hop_dur,
+        win_dur=win_dur,
+    )
+    df = interface.process_index(index)
+    df
+
+
+Serializable feature interface
+------------------------------
+
+To use a feature extractor as an input transform
+of a machine learning model
+it is recommend to provide it in a serializable way
+so it can be `stored as part of the model`_.
+One example of such a feature extractor is :class:`opensmile.Smile`.
+
+To create such a feature extractor,
+we create a class that inherits
+from :class:`audinterface.Feature`
+and :class:`audobject.Object`.
+
+.. jupyter-execute::
+
+    import audobject
+
+    class MeanStd(audinterface.Feature, audobject.Object):
+
+        def __init__(self):
+            super().__init__(
+                ['mean', 'std'],
+                process_func=self.features,
+            )
+
+        def features(self, signal, sampling_rate):
+            return [signal.mean(), signal.std()]
+
+    fex = MeanStd()
+    df = fex.process_index(index)
+    df
+
+The advantage of the feature extraction object is
+that we can save it to a YAML file
+and re-instantiate it from there.
+
+.. jupyter-execute::
+
+    fex.to_yaml('mean-std.yaml')
+    fex2 = audobject.from_yaml('mean-std.yaml')
+    df = fex2.process_index(index)
+    df
+
+
+Segmentation interface
+----------------------
+
```

Comment:
Like with the other sections I think we should provide a short introduction here.

Reply:
Oh, totally forgot that.

I now added

![image](https://user-images.githubusercontent.com/173624/173046418-999f9d08-d30d-4ded-8ad8-a29ebdc27ab1.png)

Reply:
Cool, thanks. No more comments, so I think we are good to go.